### DISCUSSION TITLE
Section 4 - Slither installation error - ERROR: rlp 1.2.0 has requirement eth-utils<2,>=1.0.2, but you'll have eth-utils 2.3.1 which is incompatible.

### DISCUSSION THREAD

**Discussion by xKeywordx** [Cyfrin/security-and-auditing-full-course-s23#67] [Q&A] [answered]
Hello guys,

I am trying to install Slither, but I keep running into those dependency conflicts. At first, I had 4 conflicts to solve and with the help of ChatGPT I was able to fix 3 of them, but right now I am stuck here and can not install Slither.

Is anyone here that ran into the same issue? How did you solve it?

The error
```
ERROR: rlp 1.2.0 has requirement eth-utils<2,>=1.0.2, but you'll have eth-utils 2.3.1 which is incompatible.
```

The conflict with rlp and eth-utils is more challenging because `rlp` requires `eth-utils` version to be less than 2, but `slither-analyzer` requires it to be greater than `2.1.0`. This is a classic case of incompatible dependencies.

Here are a few approaches to resolve this:

Option 1: Downgrade rlp to a version compatible with eth-utils 2.3.1, if such a version exists.
Option 2: Downgrade eth-utils to meet the requirements of rlp, and then check if slither-analyzer still functions correctly.

I've tried Options 1 and 2, but none of them worked because if I downgrade any of them, it will cause more errors.

**Comment by PatrickAlphaC** [comment, marked as answer]
Yeah, this is one of the annoying thing with python packages. 

Could you try installing slither with pipx? Pipx installs all packages to their own isolated virtual environment so you don't have to deal with package dependencies. I like using pipx for all my command line python installs for exactly this reason. 

Steps:
1. Install [pipx](https://github.com/pypa/pipx)
2. Use `pipx` to install slither. ie: `pipx install slither-analyzer`

**Reply by xKeywordx** [reply]
It worked. Thank you very much.
Problem solved

**Reply by giriraj-kulkarni** [reply]
i tried all the methods in this discussions  but it shows me errors like


  ERROR: Failed building wheel for lru-dict
  Running setup.py clean for lru-dict
Failed to build ckzg cytoolz lru-dict
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (ckzg, cytoolz, lru-dict)```

**Reply by giriraj-kulkarni** [reply]
not even conda works nor pipx nor changing python 3 and python nor wsl or docker
or git

**Reply by EngrPips** [reply]
Thats tough. Maybe check Slither's official GitHub  https://github.com/crytic/slither for help.
